small changes to calculations of transfers

ChristianZimpelmann opened the pull request:
### What problem do you want to solve?

Several additional small changes to the calculation of transfers that came up when comparing the result of example households:

- Add missing/wrong parameters in `eink_st_abzuege_params["einführungsfaktor`, `soz_vers_beitr_params["arbeitsl_v"]`, and `soz_vers_beitr_params["beitr_bemess_grenze_m"]`.
- Improve documentation
- Make sure that dependent children do not receive individual level Wohngeld subtractions (this is needed as long they are still part of the same tax unit as their parents)
- Improve readability of `_arbeitsl_geld_2_berechtigte_wohnfläche_hh`
- rename `unterhalt.py` to `unterhaltsv.py`

### Todo

- [x] Implement changes
- [x] Adjust tests

hmgaudecker commented:
> Note: `main` not merged yet. This PR contains changes of #458 and #461 so maybe easier to merge main into this PR.

Luckily, Git was smart enough to handle that. So we should merge those two first and then this one.

ChristianZimpelmann commented:
I made a few final changes and updated the list of changes:

- Add missing/wrong parameters in `eink_st_abzuege_params["einführungsfaktor`, `soz_vers_beitr_params["arbeitsl_v"]`, and `soz_vers_beitr_params["beitr_bemess_grenze_m"]`.
- Improve documentation
- Make sure that dependent children do not receive individual level Wohngeld subtractions (this is needed as long they are still part of the same tax unit as their parents)
- Improve readability of `_arbeitsl_geld_2_berechtigte_wohnfläche_hh`
- rename `unterhalt.py` to `unterhaltsv.py`

@LauraGergeleit, would be great if you can check the first four points whether they make sense to you and whether sources are complete, understandable, and correct.
@hmgaudecker, I also added you as reviewer.. especially for the last change.

hmgaudecker commented on src/_gettsim/parameters/eink_st_abzuege.yaml:
```diff
@@ -100,8 +100,8 @@ vorsorge2004_vorwegabzug:
   description:
     de: >-
       § 10 (4a) EStG. Früher §10 (3) EStG. Hier steht jeweils der Wert für den
-      Steuerpflichtigen. Wird verdoppelt für gemeinsam veranlagte Ehegatten. Der gesamte
```
Can you revert, please? Else this will just be going back and forth down the road...

hmgaudecker commented on src/_gettsim/parameters/soz_vers_beitr.yaml:
```diff
@@ -7,14 +7,14 @@ beitr_satz:
     de:
       ges_krankenv: >-
         Beitragssätze für gesetzliche Krankenversicherung. mean_allgemein -
-        durchschnittlicher Beitragssatz, bis 2008 allgemein - allgemeiner Satz (§241 SGB
```
See above.

hmgaudecker commented on src/_gettsim/parameters/eink_st_abzuege.yaml:
```diff
@@ -408,11 +408,13 @@ einführungsfaktor:
     en: Introductory factor for pension expense
   description:
     de: >-
-      In den Jahren 2005-2025 steigt der Anteil der abzugsfähigen Beiträge von 60% in
```
I think it should be clear from the docstring that this was a discrete change to a pre-announced path (see [comment](https://github.com/iza-institute-of-labor-economics/gettsim/issues/491#issuecomment-1400276767) in #491) -- whether we actually do anything with that, we shall see.

ChristianZimpelmann replied:
good point. Done.

hmgaudecker reviewed (changes requested):
Excellent, many thanks! Just a couple of small textual changes I'd like to see.

LauraGergeleit reviewed (commented):
The changes look good to me and are well documented!

hmgaudecker commented on src/_gettsim_tests/test_vorsorgeaufw.py:
```diff
@@ -23,7 +23,7 @@
 OUT_COLS = ["vorsorgeaufw_tu"]
 
-YEARS = [2004, 2005, 2010, 2018, 2020, 2021, 2022, 2024, 2025]
+YEARS = [2004, 2005, 2010, 2018, 2020, 2021, 2022, 2025]
```
```suggestion
YEARS = [2004, 2005, 2010, 2018, 2020, 2021, 2022, 2023]
```
? 

Seems important to make sure the discrete change is caught correctly at the right time?

hmgaudecker reviewed (approved):
I think I would like to see the test year changed, but maybe you have good reasons to pick 2025.

Great work!

hmgaudecker commented on src/_gettsim_tests/test_sozialv_beitr.py:
```diff
@@ -23,7 +23,7 @@
 
 
-YEARS = [2002, 2010, 2018, 2019, 2020, 2022, 2023]
+YEARS = ["2002", "2010", "2018", "2019", "2020", "2022", "2022-10"]
```
Should we add an issue to add tests for 2023?

hmgaudecker reviewed (approved):
Looks great! Anything missing at this moment? Else I would go ahead and branch 0.6.
